src/Tools/BuildValidator/Program.cs

```diff
@@ -134,6 +152,57 @@ static int HandleCommand(string[] assembliesPath, string sourcePath, string[]? r
             }
         }
 
+        private static AssemblyInfo[] GetAssemblyInfos(
+            IEnumerable<string> assemblySearchPaths,
+            IEnumerable<string> excludes,
+            ILogger logger)
+        {
+            var map = new Dictionary<Guid, AssemblyInfo>();
+            foreach (var directory in assemblySearchPaths)
+            {
+                foreach (var filePath in getAssemblyPaths(directory))
+                {
+                    if (excludes.Any(x => filePath.IndexOf(x, FileNameEqualityComparer.StringComparison) >= 0))
+                    {
+                        logger.LogInformation($"Skipping excluded file {filePath}");
+                        continue;
+                    }
+
+                    if (Util.GetPortableExecutableInfo(filePath) is not { } peInfo)
+                    {
+                        logger.LogError($"Skipping non-pe file {filePath}");
+                        continue;
+                    }
+
+                    if (peInfo.IsReadyToRun)
+                    {
+                        logger.LogError($"Skipping ReadyToRun file {filePath}");
+                        continue;
+                    }
+
+                    if (map.TryGetValue(peInfo.Mvid, out var assemblyInfo))
+                    {
+                        // It's okay for the assembly to be duplicated in the search path.
+                        logger.LogInformation("Duplicate assembly path have same MVID");
+                        logger.LogInformation($"\t{filePath}");
+                        logger.LogInformation($"\t{assemblyInfo.FilePath}");
+                        continue;
+                    }
+
+                    map[peInfo.Mvid] = new AssemblyInfo(filePath, peInfo.Mvid);
+                }
+            }
+
+            return map.Values.OrderBy(x => x.FileName).ToArray();
```

Comment:
>x => x.FileName [](start = 38, length = 15)

Should this use `FileNameEqualityComparer.StringComparison`? Then there would be no need to sort the values again in `logAssemblyInfos()`.

Reply:
makes sense